File: src/main/erlang/g0101_0200/s0136_single_number/Solution.erl
% #Easy #Top_100_Liked_Questions #Top_Interview_Questions #Array #Bit_Manipulation
% #Data_Structure_II_Day_1_Array #Algorithm_I_Day_14_Bit_Manipulation #Udemy_Integers
% #Top_Interview_150_Bit_Manipulation #Big_O_Time_O(N)_Space_O(1)
% #2025_01_16_Time_0_ms_(100.00%)_Space_72.99_MB_(100.00%)

-spec single_number(Nums :: [integer()]) -> integer().
single_number(Nums) ->
    %% using lists:foldl to calculate the xor of all elements in the list
    lists:foldl(fun(X, Acc) -> X bxor Acc end, 0, Nums).

File: src/main/go/g0101_0200/s0136_single_number/solution.go
package s0136_single_number

// #Easy #Top_100_Liked_Questions #Top_Interview_Questions #Array #Bit_Manipulation
// #LeetCode_75_Bit_Manipulation #Data_Structure_II_Day_1_Array
// #Algorithm_I_Day_14_Bit_Manipulation #Udemy_Integers #Top_Interview_150_Bit_Manipulation
// #Big_O_Time_O(N)_Space_O(1) #2025_05_07_Time_0_ms_(100.00%)_Space_7.82_MB_(83.54%)

func singleNumber(nums []int) int {
	res := nums[0]
	for i := 1; i < len(nums); i++ {
		res = res ^ nums[i]
	}
	return res
}
